dallinger/deployment.py

```diff
@@ -26,37 +26,44 @@
     setup_experiment,
 )
 
-INITIAL_DELAY = 1
+DEFAULT_DELAY = 1
 BACKOFF_FACTOR = 2
 MAX_ATTEMPTS = 6
 
 
-def _handle_launch_data(url, error, delay=INITIAL_DELAY, attempts=MAX_ATTEMPTS):
+def handle_launch_data(url, error, delay=DEFAULT_DELAY, attempts=MAX_ATTEMPTS):
+    """Sends a POST request to te given `url`, retrying it with exponential backoff.
+    The passed `error` function is invoked to give feedback as each error occurs,
+    possibly multiple times.
+    """
     launch_data = None
     launch_request = None
     for remaining_attempt in sorted(range(attempts), reverse=True):  # [3, 2, 1, 0]
-        time.sleep(delay)
         try:
             launch_request = requests.post(url)
             request_happened = True
-        except requests.exceptions.RequestException:
+        except requests.exceptions.RequestException as err:
             request_happened = False
-            error(f"Error accessing {url}")
+            error(f"Error accessing {url}:\n{err}")
 
-        try:
-            if request_happened:
+        if request_happened:
+            try:
                 launch_data = launch_request.json()
-        except ValueError:
-            error(
-                "Error parsing response from {}, "
-                "check web dyno logs for details: {}".format(url, launch_request.text)
-            )
-            raise
-        except json.decoder.JSONDecodeError:
-            # The Heroku backend did not return JSON. It means our dallinger instance
-            # was not (yet) running at the time of the request.
-            # We treat this similarly to a RequestException: we'll try again after waiting.
-            request_happened = False
+            except json.decoder.JSONDecodeError:
+                # The backend did not return JSON. It means our dallinger instance
+                # was not (yet) running at the time of the request.
+                # We treat this similarly to a RequestException: we'll try again after waiting.
+                request_happened = False
+                error(
+                    f"Error parsing response from {url}, "
+                    f"check logs for details.\n{launch_request.text}"
+                )
+            except ValueError as err:
+                error(
+                    f"Error parsing response from {url}, "
+                    f"check logs for details.\n{err}\n{launch_request.text}"
```

Comment:
can we say `server logs` instead of just `logs`?

Reply:
Good point! Done.